### PR TITLE
656 filter by identifier

### LINKED ISSUE #656: Quest v0.2/ G6PD: Add the ability to search for patients by identifier
**Is your feature request related to a problem? Please describe.**
1. Current search allows you to only search by name (given & family name). We need to add the ability to search for users by their given Identifier eg. 111000111 based on the sample snippet given below 

**Describe the solution you'd like**
1. Add the ability to enter and filter patients based on their given identifier 
**Additional context**
1. This may need SDK support so a separate ticket to be created/tracked there

**Sample patient** 
https://turn-fhir.smartregister.org/fhir/Patient/969
```
{
  "resourceType": "Patient",
  "id": "969",
  "identifier": [
    {
      "use": "official",
      "system": "http://who.int/ddcc/hcid",
      "value": "111000111"
    }
  ],
  "active": true,
  "name": [
    {
      "family": "Murphy",
      "given": [
        "Eddie"
      ]
    }
  ],
  "telecom": [
    {
      "system": "phone",
      "value": "744585"
    }
  ],
  "gender": "male",
  "birthDate": "1986-09-19",
  "address": [
    {
      "city": "Rome",
      "country": "Italy"
    }
  ]
}
```

### PR DESCRIPTION
**Pull request by Rkareko**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #[issue number] 

Fixes #656 

**Type**
Choose one: (Bug fix | Feature | Documentation | Testing | Code health | Release | Other) 

**Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [x] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/quest/src/test/java/org/smartregister/fhirecore/quest/ui/patient/register/PatientRegisterFragmentTest.kt [hunk 1]
```diff
@@ -74,6 +74,14 @@ class PatientRegisterFragmentTest : RobolectricTest() {
...
       )
     )
+
+    Assert.assertTrue(
+      registerFragment.performFilter(
+        RegisterFilterType.SEARCH_FILTER,
+        PatientItem(id = "1234"),
+        "1234"
+      )
+    )
```
Comment (ellykits) [inline android/quest/src/test/java/org/smartregister/fhirecore/quest/ui/patient/register/PatientRegisterFragmentTest.kt:84]: Flip the actual and expected result for this assertion.

**Review by ellykits — CHANGES_REQUESTED** [review]

**Comment by ndegwamartin** [comment]
@Rkareko Please update the **CHANGELOG.md** file and the _PR template check list_ as well

File: android/quest/src/test/java/org/smartregister/fhirecore/quest/ui/patient/register/PatientRegisterFragmentTest.kt (hunk 1 above)
Comment (Rkareko) [reply android/quest/src/test/java/org/smartregister/fhirecore/quest/ui/patient/register/PatientRegisterFragmentTest.kt:84]: @ellykits This is an `assertTrue` statement which takes in one argument.

Comment (ellykits) [reply android/quest/src/test/java/org/smartregister/fhirecore/quest/ui/patient/register/PatientRegisterFragmentTest.kt:84]: Haha damn me! :joy: Noted approved.

**Review by ellykits — APPROVED** [review]

**Pull request merged**